leku/src/main/java/com/schibstedspain/leku/search/LekuSearchFragment.java

```diff
@@ -0,0 +1,252 @@
+      }
+      return false;
+    }
+    return true;
+  }
+
+  private void setUpClearListener() {
+    clearSearchButton.setOnClickListener(view -> searchView.setText(""));
+  }
+
+  private void setUpSearchViewListeners() {
+    searchView.setOnEditorActionListener((v, actionId, event) -> {
+      boolean handled = false;
+      if (actionId == EditorInfo.IME_ACTION_SEARCH) {
+        retrieveLocationFrom(v.getText().toString());
+        closeKeyboard();
+        handled = true;
+      }
+      return handled;
+    });
+    textWatcher = getSearchTextWatcher();
+    searchView.addTextChangedListener(textWatcher);
+  }
+
+  private void closeKeyboard() {
+    InputMethodManager imm = (InputMethodManager) getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
+    imm.hideSoftInputFromWindow(searchView.getWindowToken(), 0);
+  }
+
+  @Override
+  public void onViewStateRestored(@Nullable Bundle savedInstanceState) {
+    super.onViewStateRestored(savedInstanceState);
+    if (savedInstanceState != null) {
+      String lastQuery = savedInstanceState.getString(LAST_LOCATION_QUERY, "");
+      if (!"".equals(lastQuery)) {
+        retrieveLocationFrom(lastQuery);
+      }
+    }
+  }
+
+  private void retrieveLocationFrom(String query) {
+    retrieveLocationFrom(query, false);
+  }
+
+  private void retrieveLocationFrom(String query, boolean debounce) {
+    lekuSearchCallback.retrieveLocationFrom(query, debounce);
+  }
+
+  @Override
+  public void onSaveInstanceState(Bundle outState) {
+    super.onSaveInstanceState(outState);
+    outState.putString(LAST_LOCATION_QUERY, String.valueOf(searchView.getText()));
+  }
+
+  @Override
+  public void onDestroy() {
+    if (searchView != null && textWatcher != null) {
+      searchView.removeTextChangedListener(textWatcher);
+    }
+    super.onDestroy();
+  }
+
+  private TextWatcher getSearchTextWatcher() {
+    return new LekuSearchTextWatcher(MIN_CHARACTERS) {
+      @Override
+      void onEmptyText() {
+        if (clearSearchButton != null) {
+          clearSearchButton.setVisibility(View.INVISIBLE);
+        }
+        if (searchOption != null) {
+          searchOption.setIcon(R.drawable.ic_mic);
+        }
+        clearSearchResults();
+        showLocationInfoLayout();
+      }
+
+      @Override
+      void onText(CharSequence charSequence) {
+        if (clearSearchButton != null) {
+          clearSearchButton.setVisibility(View.VISIBLE);
+        }
+        if (searchOption != null) {
+          searchOption.setIcon(R.drawable.ic_search);
+        }
+        retrieveLocationFrom(charSequence.toString(), true);
+      }
+    };
+  }
+
+  private void clearSearchResults() {
+    lekuSearchCallback.clearSearchResults();
+  }
+
+  private void showLocationInfoLayout() {
+    lekuSearchCallback.showLocationInfoLayout();
+  }
+
+  @Override
+  public void onActivityResult(int requestCode, int resultCode, Intent data) {
+    switch (requestCode) {
```

Comment:
Why not a simpler `if` clause?
`if (resultCode == Activity.RESULT_OK && requestCode == REQUEST_PLACE_PICKER) { ...`
I think its easier to read. Having that break; default: break; is expressing that there's more code than needed, IMHO. Not a big deal, btw!